src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -192,7 +197,146 @@ private DenseMatrix initRhs(LfNetwork lfNetwork, EquationSystem equationSystem,
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Balance type not yet supported: " + loadFlowParameters.getBalanceType());
+            }
+        }
+        return participationList;
+    }
+
+    /**
+     * Create a map containing sensibility values, indexed first by FunctionId, then by factorId
+     *
+     * @param sensitivityValues
+     * @param remainder
+     * @return
+     */
+    public static Map<String, Map<String, SensitivityValue>> convertSensitivityValuesToMap(List<SensitivityValue> sensitivityValues, List<SensitivityValue> remainder) {
+        Map<String, Map<String, SensitivityValue>> sensitivityValuesMap = new HashMap<>();
+        for (SensitivityValue sensitivityValue : sensitivityValues) {
+            SensitivityFactor factor = sensitivityValue.getFactor();
+            if (factor instanceof BranchFlowPerInjectionIncrease) {
+                Map<String, SensitivityValue> branchSentivityMap = sensitivityValuesMap.computeIfAbsent(factor.getFunction().getId(), branch -> new HashMap<>());
+                branchSentivityMap.put(factor.getVariable().getId(), sensitivityValue);
+            } else {
+                remainder.add(sensitivityValue); // todo: if this is not an injection, I do not know how to do
+            }
+        }
+        return sensitivityValuesMap;
+    }
+
+    public List<SensitivityValue> correctSensitivityValues(List<SensitivityValue> sensitivityValues, LfNetwork lfNetwork, LoadFlowParameters lfParameters) {
+
+        // if the slack is distributed, we need to compute sensitivities corrected from the distribution effects.
+        List<SensitivityValue> correctedSensitivityValues = new ArrayList<>();
+
+        // map indexed by functionid, then variableid
+        Map<String, Map<String, SensitivityValue>> sensitivityValuesMap = convertSensitivityValuesToMap(sensitivityValues, correctedSensitivityValues);
+
+        List<Pair<String, Number>> participatingElements = getParticipatingElements(lfNetwork, lfParameters);
+
+        // compute the slack distribution correction by branch
+        Map<String, Number> slackDistributionCorrection = new HashMap<>();
+        for (String lfBranchId : sensitivityValuesMap.keySet()) {
+            Map<String, SensitivityValue> sensitivitiesByBranch = sensitivityValuesMap.get(lfBranchId);
+            double sum = participatingElements.stream()
+                    .mapToDouble(element -> element.getRight().doubleValue() * sensitivitiesByBranch.get(element.getLeft()).getValue())
+                    .sum();
+            slackDistributionCorrection.put(lfBranchId, sum);
+        }
+
+        // compute sensitivities corrected from the slack distribution effects
+        for (SensitivityValue sensitivityValue : sensitivityValues) {
```

Comment:
Here a fix is needed because I don't have sort the sensitivity values to give only those that have been ask in the API. @Djazouli your code was functionaly right but it was to remove the use of factors.